aarch64-linux: foundationdb-simulation fails to compile — `*const i8` should be `c_char` (str_from_c + register macros)

## Summary

`foundationdb-simulation` (0.2.3, and `main` @ HEAD) does **not compile for `aarch64-unknown-linux-gnu`**. Several FFI signatures hardcode `*const i8` where the C `char *` maps to `*const c_char` — and `c_char` is **unsigned (`u8`) on aarch64‑Linux** (it is `i8` on macOS, incl. Apple‑Silicon, and on `x86_64-unknown-linux-gnu`, which is why those targets build fine).

## Environment

- `foundationdb-simulation = "0.2.3"` (also reproduces on `main` @ HEAD), feature `fdb-7_4`
- target: `aarch64-unknown-linux-gnu` (Debian bookworm, `rustc 1.85.1`)
- builds OK on `x86_64-unknown-linux-gnu` and macOS arm64/x86_64 (`c_char == i8` there)

## Errors

```
error[E0308]: mismatched types
  --> .../foundationdb-simulation-0.2.3/src/bindings.rs:31:46
   |  let c_str = unsafe { ffi::CStr::from_ptr(c_buf) };
   |                       ------------------- ^^^^^ expected `*const u8`, found `*const i8`
   = note: expected raw pointer `*const u8`
              found raw pointer `*const i8`
   = note: CStr::from_ptr expects `*const c_char` (== `*const u8` on aarch64-linux)

error[E0308]: mismatched types
   --> .../foundationdb-simulation-0.2.3/src/bindings.rs:168:32
    |  let value = str_from_c(raw_value.inner);
    |              ---------- ^^^^^^^^^^^^^^^ expected `*const i8`, found `*const u8`
    = note: expected raw pointer `*const i8`
               found raw pointer `*const u8`

error: could not compile `foundationdb-simulation` (lib) due to 2 previous errors
```

## Root cause

`src/bindings.rs:30` declares the param as `*const i8`:

```rust
pub fn str_from_c(c_buf: *const i8) -> String {
    let c_str = unsafe { ffi::CStr::from_ptr(c_buf) };  // from_ptr wants *const c_char
    c_str.to_str().unwrap().to_string()
}
```

On aarch64‑Linux `c_char == u8`, so (a) `CStr::from_ptr(c_buf)` rejects the `*const i8` arg, and (b) the call site at `src/bindings.rs:168` passes `raw_value.inner: *const u8` (the bindgen‑mapped `c_char`) into `str_from_c(*const i8)`. The `register_factory!` / `register_workload!` macros (`src/lib.rs:204` and `:240`) likewise declare the C entry points as `extern "C" fn workloadCFactory(raw_name: *const i8, …)`.

## Suggested fix

Use `c_char` (from `core::ffi` / `std::os::raw`) at the FFI boundary instead of the hardcoded `i8`. It resolves to `i8` on the currently‑working targets (no change there) and to `u8` on aarch64‑Linux:

- `src/bindings.rs:30` → `pub fn str_from_c(c_buf: *const c_char) -> String`
- `src/lib.rs:204` & `:240` → `extern "C" fn workloadCFactory(raw_name: *const c_char, …)`

Happy to open a PR if that's helpful.

## Reproducer

Build any external‑workload `cdylib` that uses `register_factory!` / `register_workload!` — or simply the crate itself — on an aarch64 Linux host (or for `--target aarch64-unknown-linux-gnu`):

```
cargo build -p foundationdb-simulation --features fdb-7_4   # on aarch64-linux
```

--- a/foundationdb-simulation/src/bindings.rs
+++ b/foundationdb-simulation/src/bindings.rs
@@ -3,7 +3,11 @@
 //! This module defines all C and Rust structures.
 //! It also provides bindings and wrappers to map behavior from Rust to C.
 
-use std::{ffi, str::FromStr, time::Duration};
+use std::{
+    ffi::{self, c_char},
+    str::FromStr,
+    time::Duration,
+};
 
 use foundationdb as fdb;
 
@@ -32,7 +36,7 @@ pub const FDB_WORKLOAD_API_VERSION: i32 = raw_bindings::FDB_WORKLOAD_API_VERSION
 
 #[doc(hidden)]
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
-pub fn str_from_c(c_buf: *const i8) -> String {
+pub fn str_from_c(c_buf: *const c_char) -> String {
     let c_str = unsafe { ffi::CStr::from_ptr(c_buf) };
     c_str.to_str().unwrap().to_string()
 }

--- a/foundationdb-simulation/src/lib.rs
+++ b/foundationdb-simulation/src/lib.rs
@@ -224,7 +224,7 @@ macro_rules! register_factory {
     ($name:ident) => {
         #[unsafe(no_mangle)]
         extern "C" fn workloadCFactory(
-            raw_name: *const i8,
+            raw_name: *const std::ffi::c_char,
             raw_context: $crate::internals::FDBWorkloadContext,
         ) -> $crate::WrappedWorkload {
             use std::sync::atomic::{AtomicBool, Ordering};
@@ -260,7 +260,7 @@ macro_rules! register_workload {
     ($name:ident) => {
         #[unsafe(no_mangle)]
         extern "C" fn workloadCFactory(
-            raw_name: *const i8,
+            raw_name: *const std::ffi::c_char,
             raw_context: $crate::internals::FDBWorkloadContext,
         ) -> $crate::WrappedWorkload {
             use std::sync::atomic::{AtomicBool, Ordering};

